ann_benchmarks/results.py

```diff
@@ -16,13 +16,13 @@ def get_result_filename(dataset=None, count=None, definition=None,
         d.append(str(count))
     if definition:
         d.append(definition.algorithm + ('-batch' if batch_mode else ''))
-        if 'redisearch' in definition.algorithm:
-            prefix = re.sub(r'\W+', '_', json.dumps(query_arguments, sort_keys=True)).strip('_')
-            d.append(prefix + f'_client_{id}.hdf5')
-        else:
-            data = definition.arguments + query_arguments
-            d.append(re.sub(r'\W+', '_', json.dumps(data, sort_keys=True))
-                     .strip('_') + ".hdf5")
+        data = definition.arguments + query_arguments
+        for i in range(len(data)):
+            if isinstance(data[i], dict):
+                data[i] = {k:data[i][k] for k in data[i] if data[i][k] is not None}
+        data.append('client')
+        data.append(id)
+        d.append(re.sub(r'\W+', '_', json.dumps(data, sort_keys=True)).strip('_') + ".hdf5")
```

Comment:
what is the purpose of this modification?

Reply:
have cleaner names for the output files. Before:
`angular_user_null_auth_null_host_null_port_null_cluster_false_HNSW_M_32_efConstruction_500_20_client_1.hdf5`
After:
`angular_cluster_false_HNSW_M_32_efConstruction_500_20_client_1.hdf5`